### PR TITLE
Trace reports - exchange-specific tracing; proof exchange tracing

### PR DESCRIPTION
**Pull request by ianco**
Still a few outstanding TODO's, but opening up the PR to get some eyes on it.

Adds exchange-specific tracing (adds a "trace" attribute to the exchange records as well as admin API inputs and outputs).

Added tracing to proof exchange and credential exchange.

Added support to the alice/faber demo to support tracing to standard log and http endpoint.

TODO:
- append trace logs to messages is partially implemented - need to either finish this or rip out the code
- had to add reference to the EFK docker network into the test and demo scripts - need to clean this up - FIXED
- also need to look into the implications for message forwarding and routing ...
- anything else from the reviews ...

### CODE REVIEW

**Comment by ianco** [comment]
> This pull request **introduces 1 alert** when merging [79f3254](https://github.com/hyperledger/aries-cloudagent-python/commit/79f3254fde2f624d353c57dba284fe6a867aa3ee) into [caf4572](https://github.com/hyperledger/aries-cloudagent-python/commit/caf457276b19df374c16c2890e1c7e4914f46254) - [view on LGTM.com](https://lgtm.com/projects/g/hyperledger/aries-cloudagent-python/rev/pr-ddc015f4a85283a0f81fcde644c038fe9e287656)
> 
> **new alerts:**
> 
> * 1 for __eq__ not overridden when adding attributes

Fixed

File: demo/runners/faber.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def __init__(
         self, http_port: int, admin_port: int, no_auto: bool = False, **kwargs
     ):
         super().__init__(
-            "Faber Agent",
+            "Faber.Agent",
```
Comment (andrewwhitehead) [inline demo/runners/faber.py:39]: Not sure why it's renaming the demo agents?

Comment (ianco) [reply demo/runners/faber.py:39]: When I was outputting the data from Kibana to Excel I had to use space delimiters and this was messing up the formatting.  (csv export wasn't available on the eval version of Kibana)

Comment (andrewwhitehead) [reply demo/runners/faber.py:39]: Mainly wondering if that would require changes to documentation. @swcurran ?

Comment (ianco) [reply demo/runners/faber.py:39]: We can change it back if it's a big deal ...

**Comment by ianco** [comment]
Will likely have further changes but I'd like to get this PR merged and into a release (even a beta release) so we can get this deployed/tested on OpenShift

**Pull request merged**